### PR TITLE
feat: add /update-skills CRM slash command (#47)

### PR DESCRIPTION
**Pull request by happysmile001**
Add Discord command to update structured skills (cSkillAttrs) with strength ratings 1-5. Members can update their own skills; Steering Committee+ can update skills for other users. New skills are merged with existing ones, preserving unchanged entries.

All 7 integration tests pass. Here's what was verified end-to-end:

Test	What it verifies
1. Basic parsing	"python:5, react:4, go:3" parses correctly
2. Alias normalization	Node.JS→node, js→javascript, k8s→kubernetes, py→python
3. Edge cases	Trailing commas, extra spaces, duplicate skills (last wins)
4. Rejection cases	Empty input, missing colon, empty skill name, strength 0/6/-1/abc
5. Merge + roundtrip	Existing skills preserved, new added, overlapping updated, JSON serialization roundtrips correctly
6. Full command flow	Defer → search contact → fetch → merge → PUT → embed response with * markers on updated skills
7. Permission denied	Member role blocked from updating others, no API call made

## Summary by CodeRabbit

* **New Features**
  * Added an /update-skills slash command to add or merge comma-separated skill:strength entries, resolve target contacts, enforce role-based permission for updating others, and return a formatted confirmation with a CRM profile link.
  * Improved input parsing, normalization, and validation with clear user feedback and audit events.

* **Error Handling**
  * Handles API and unexpected errors with logging, auditing, and user-facing messages; resilient audit logging.

* **Tests**
  * Added extensive unit tests covering parsing, normalization, self/other updates, permission checks, merging, and error cases.

### CODE REVIEW

File: apps/discord_bot/src/five08/discord_bot/cogs/crm.py [hunk 1]
```diff
@@ -3140,6 +3177,219 @@ async def view_skills(
...
+                merged.items(), key=lambda item: (-item[1], item[0].casefold())
+            ):
+                marker = " *" if skill in new_skills else ""
+                updated_lines.append(f"- `{skill}` ({strength}/5){marker}")
+            if len(updated_lines) > 25:
+                updated_lines = updated_lines[:25]
+                updated_lines.append(f"...and {len(merged) - 25} more.")
+            embed.add_field(
+                name="Skills", value="\n".join(updated_lines), inline=False
+            )
```
[the remark this answers is not in the document]
Comment (happysmile001) [reply apps/discord_bot/src/five08/discord_bot/cogs/crm.py]: https://github.com/508-dev/508-workflows/pull/60/changes/1b48a2af7a8a2883f12164cebeb5ea9fb6912c68

File: apps/discord_bot/src/five08/discord_bot/cogs/crm.py [hunk 2]
```diff
@@ -3140,6 +3220,222 @@ async def view_skills(
...
+                result="success",
+                metadata={
+                    "search_term": query or None,
+                    "target_scope": target_scope,
+                    "skills_updated": list(new_skills.keys()),
+                    "skills_count": len(merged),
+                },
+                resource_type="crm_contact",
+                resource_id=contact_id,
+            )
```
[the remark this answers is not in the document]
Comment (happysmile001) [reply apps/discord_bot/src/five08/discord_bot/cogs/crm.py]: https://github.com/508-dev/508-workflows/pull/60/commits/1dd798168ce6911a29963d2638e88198d1e95c07
